app/main.py

```diff
@@ -2,0 +1,30 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        print(f"Animal __init__ called with: {name}, health: {health}")
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def die(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, Hidden: {self.hidden}}}")
```

Comment:
Code Style issue: Use interpolation instead of concatenation. Here is the corrected code: `return f"{{Name: {self.name}, Health: {self.health}, Hidden: {self.hidden}}}"`.

Reply:
done